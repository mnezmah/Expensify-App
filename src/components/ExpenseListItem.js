import React from 'react';
import expenses from '../selectors/expenses';

const ExpenseListItem = (props) => (
  <div>
      {expenses.description}
      {expenses.amount}
      {expenses.createdAt}
  </div>
);

export default ExpenseListItem;

//export a stateless functional component
//that component will render: 
//  - description
//  - amount
//  -  createdAt value
// thopse three above are props passed in
// use map() - peek in indecision code
// Water bill and Gass bill need to fbe shown on the screen instead of 2 :)
// you have to render many instances of expenseList item here
// fill out the expenseList Item component file