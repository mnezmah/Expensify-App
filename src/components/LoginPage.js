<<<<<<< HEAD
import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth.js';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <h3>Login section</h3>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
=======
import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control.</p>
      <button className="button" onClick={startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
>>>>>>> 314442f1bb91add33cd0a720d0a80d9dd71895d8
