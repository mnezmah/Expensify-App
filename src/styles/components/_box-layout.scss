.box-layout {
<<<<<<< HEAD
  background:url('/images/bg.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
}
=======
  align-items: center;
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, .15);
  border-radius: 3px;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
>>>>>>> 314442f1bb91add33cd0a720d0a80d9dd71895d8
