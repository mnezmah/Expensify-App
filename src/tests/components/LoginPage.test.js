import React from 'react';
import { shallow, configure } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test ('should render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});