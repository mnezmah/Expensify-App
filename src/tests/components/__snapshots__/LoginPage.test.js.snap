// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<div>
  <h3>
    Login section
  </h3>
  <button>
    Login
  </button>
</div>
`;
